ci/generate-and-upload-build-steps.sh

```diff
@@ -2,59 +2,80 @@
 set -euo pipefail
 
 upload_build_configuration_step() {
-    export ENGINE_COMMIT_HASH=$1
-    export BUILD_PLATFORM=$2
-    export BUILD_TARGET=$3
-    export BUILD_STATE=$4
-    buildkite-agent pipeline upload "ci/gdk_build.template.steps.yaml"
+    export ENGINE_COMMIT_HASH="${1}"
+    export BUILD_PLATFORM="${2}"
+    export BUILD_TARGET="${3}"
+    export BUILD_STATE="${4}"
+
+    if [[ ${BUILD_PLATFORM} == "Mac" ]]; then
+        export BUILD_COMMAND="./ci/setup-build-test-gdk.sh"
+        REPLACE_STRING="s|BUILDKITE_AGENT_PLACEHOLDER|macos|g;"
+    else
+        export BUILD_COMMAND="powershell ./ci/setup-build-test-gdk.ps1"
+        REPLACE_STRING="s|BUILDKITE_AGENT_PLACEHOLDER|windows|g;"
+    fi
+
+    sed "$REPLACE_STRING" "ci/gdk_build.template.steps.yaml" | buildkite-agent pipeline upload
 }
 
 generate_build_configuration_steps () {
     # See https://docs.unrealengine.com/en-US/Programming/Development/BuildConfigurations/index.html for possible configurations 
     ENGINE_COMMIT_HASH="${1}"
 
-    # if the BUILD_ALL_CONFIGURATIONS environment variable doesn't exist, then...
-    if [[ -z "${BUILD_ALL_CONFIGURATIONS+x}" ]]; then
-        echo "Building for subset of supported configurations. Generating the appropriate steps..."
-        
-        # Win64 Development Editor build configuration
-        upload_build_configuration_step "${ENGINE_COMMIT_HASH}" "Win64" "Editor" "Development"
-
-        # Linux Development NoEditor build configuration
-        upload_build_configuration_step "${ENGINE_COMMIT_HASH}" "Linux" "" "Development"
+    if [[ -n "${MAC_BUILD:-}" ]]; then
```

Comment:
Nit - the mix of `-z` and `-n` for testing conditions in this block make it harder to grok